cmd/nerdctl/run.go

```diff
@@ -771,6 +771,7 @@ func generateLogURI(dataStore, logDriver string, logOptMap map[string]string) (*
 	}
 	args := map[string]string{
 		logging.MagicArgv1: dataStore,
+		logging.DriverType: logDriver,
```

Comment:
I don't think this is correct.

We have to ensure that MagicArgv1 is always argv[1], but this map may make it argv[3]

Reply:
@AkihiroSuda  not really, I have already experimented with a change of position of the args in this PR https://github.com/containerd/nerdctl/pull/959. 

the actual code
```
	args := map[string]string{
		logging.MagicArgv1: dataStore,
	}
	for k, v := range logOptMap {
		args[k] = v
	}
```
does not guarantee a fixed order of arguments
for this reason  the args parsing is refactored in this way : https://github.com/containerd/nerdctl/pull/959/files#diff-055f400ea02598bd9b9ef894b5ffc400954e88ad71cc43a31f421222af4da6c7R81